src/main/database_manager.cpp

```diff
@@ -116,18 +141,6 @@ void DatabaseManager::GetDatabaseType(ClientContext &context, string &db_type, A
 
 	// try to extract database type from path
 	if (db_type.empty()) {
-		lock_guard<mutex> write_lock(db_paths_lock);
```

Comment:
Will this work on Windows? I remember discussing Windows being much more restrictive with holding multiple file handles.

Reply:
Yes good point - I've added the check back in there